### PR TITLE
feat: Capture minidumps for hard crashes [INGEST-566]

### PR DESCRIPTION
**Pull request by jan-auer**
Uses sentry-native with the breakpad backend to offer an optional `crash-handler` feature for Linux and macOS. When enabled, it'll capture hard crashes and write minidumps with stack traces of all threads into a database folder on the file system. On next restart, those crashes will be sent to Sentry for debugging. The DSN and release are inherited from the main Sentry SDK.

Breakpad offers advantages over the other backends. First, it works on all major platforms we need to support. Second, it does not need a handler process. Third, it still captures memory dumps with all threads. In the long run, we'd probably opt for the in-house crash reporter without third-party crash reporters in pure rust.

The minidump will have to be persisted in a volume that survives container restarts. There is a new configuration `sentry._crash_db` for this.

cc @Swatinem @oioki

### CODE REVIEW

**Comment by jan-auer** [comment]
Accidentally requested review instead of opening as Draft PR. Please ignore linter and linker errors for the time being. I would like to settle on the approach first and test it out before wrapping this up.

File: relay-crash/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+pub fn init(dsn: Option<String>, release: Option<&str>) {
+    unsafe {
+        let options = sentry_options_new();
+
+        if let Some(dsn) = dsn {
+            let dsn_cstr = CString::new(dsn).unwrap();
+            sentry_options_set_dsn(options, dsn_cstr.as_ptr());
+        }
+
+        if let Some(release) = release {
```
Comment (Swatinem) [inline relay-crash/src/lib.rs]: do you need to pass `environment` here as well?

**Review by Swatinem — APPROVED** [review]
well, a few more docs would be nice ;-)

File: relay-crash/src/lib.rs (hunk 1 above)
Comment (jan-auer) [reply relay-crash/src/lib.rs]: Good point, I'll add it to the crate even though we don't set the environment in Relay.

**Review by tonyo — APPROVED** [review]
LGTM approach-wise, but someone from Owners Ingest should check the Rust bits.
Also, let's make sure that the minidumps are cleaned up after they are submitted (probably already a sentry-native default).

File: relay-crash/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                native::sentry_options_set_environment(options, env_cstr.as_ptr());
+            }
+
+            native::sentry_init(options);
+        }
+    }
+
+    #[cfg(not(unix))]
+    pub fn install(&self) {
+        unimplemented!("crash handler not supported on this platform");
```
Comment (untitaker) [inline relay-crash/src/lib.rs:87]: is it even possible to compile this module on windows?

**Review by untitaker — APPROVED** [review]

File: relay-crash/src/lib.rs (hunk 2 above)
Comment (jan-auer) [reply relay-crash/src/lib.rs:87]: it's possible, sentry-native with breakpad works on Windows, too. Given this is fully internal, I just did not build this out in the build script (requires linking against a few other libraries).

**Comment by jan-auer** [comment]
> Also, let's make sure that the minidumps are cleaned up after they are submitted (probably already a sentry-native default).

Good point. sentry-native deletes minidumps after handling them.

**Review by oioki — APPROVED** [review]

**Pull request merged**